### PR TITLE
[wip]add pinconfig and hal for gd32c10xxx

### PR DESCRIPTION
**Pull request by feilongfl**
Hi!
this PR add support for `gd32c10xxx` device and old `gd32e103xx` device.

hal code from:
http://www.gd32mcu.com/cn/download/7?kw=GD32C1
-> GD32C10x Firmware Library  `V1.0.1`

I made some modifications to hal code with reference to https://github.com/zephyrproject-rtos/hal_gigadevice/pull/6

### CODE REVIEW

File: include/dt-bindings/pinctrl/gd32c103c(b-8)xx-pinctrl.h [hunk 1]
```diff
@@ -0,0 +1,717 @@
+/*
+ * Autogenerated file
+ *
+ * SPDX-License-Identifier: Apache 2.0
+ */
+
+#include "gd32c103xx-afio.h"
```
Comment (gmarull) [inline include/dt-bindings/pinctrl/gd32c103c(b-8)xx-pinctrl.h:7]: I can't see this file in this PR

File: gd32c10x/standard_peripheral/Source/gd32c10x_can.c [hunk 2]
```diff
@@ -36,6 +36,15 @@ OF SUCH DAMAGE.
...
 #include <stdlib.h>
 
+/* CAN communication mode */
+#define CAN_NORMAL_MODE                    ((uint8_t)0x00U)             /*!< normal communication mode */
+#define CAN_LOOPBACK_MODE                  ((uint8_t)0x01U)             /*!< loopback communication mode */
+#define CAN_SILENT_MODE                    ((uint8_t)0x02U)             /*!< silent communication mode */
+#define CAN_SILENT_LOOPBACK_MODE           ((uint8_t)0x03U)             /*!< loopback and silent communication mode */
+
+/* CAN timeout */
+#define CAN_TIMEOUT                        ((uint32_t)0x0000FFFFU)      /*!< timeout value */
```
Comment (gmarull) [inline gd32c10x/standard_peripheral/Source/gd32c10x_can.c:46]: this definitions need to be public, see https://github.com/zephyrproject-rtos/hal_gigadevice/blob/main/gd32f4xx/standard_peripheral/include/gd32f4xx_can.h#L597

**Review by gmarull — CHANGES_REQUESTED** [review]

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @feilongfl .
You need review `dt-bindings: add pinctrl gd32c103xx generated headers` commit. There is mention to gd32c**4**03xx MCU.

This can only be merged after a board be build at main repository,

**Comment by gmarull** [comment]
stale

**Pull request closed without merging**